grpc/dataplane_grpc.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package config;
+
+option go_package = "dataplane/grpc"
+
+service ConfigService {
+    rpc GetGlobalConfig (GetGlobalConfigRequest) returns (GlobalConfig);
+    rpc UpdateGlobalConfig (UpdateGlobalConfigRequest) returns (UpdateGlobalConfigResponse);
+    rpc CheckConfigGeneration (CheckConfigGenerationRequest) returns (CheckConfigGenerationResponse);
+}
+
+message GetGlobalConfigRequest {}
+
+message UpdateGlobalConfigRequest {
+    GlobalConfig config = 1;
+}
+
+message UpdateGlobalConfigResponse {
+    bool status = 1;
+    string message = 2;
+}
+
+message CheckConfigGenerationRequest {}
+
+message CheckConfigGenerationResponse {
+    uint64 generation = 1;
+}
+
+message Subnet {
+    string name = 1;
+    string prefix = 2;
+    uint32 vni = 3;
+}
+
+message VRF {
+    string name = 1;
+    repeated Subnet subnets = 2;
+}
+
+message Interface {
+    string name = 1;
+    uint32 index = 2;
+    string type = 3;
+    string address = 4;
+    string pci = 5;
+    uint32 vni = 6;
+}
+
+message PeeringIPs {
+    repeated string cidr = 1;
+    repeated string not = 2;
+}
+
+message PeeringAs {
+    repeated string cidr = 1;
+    repeated string not = 2;
+}
+
+message PeeringEntry {
+    PeeringIPs ips = 1;
+    PeeringAs as = 2;
+}
+
+message Peering {
+    string name = 1;
+    map<string, PeeringEntry> entries = 2;
+}
+
+message GlobalConfig {
```

Comment:
just a question - why not maps for VRFs(VPCs)/Peerings?

Reply:
you mean using entire VPC as a key and not have it separately delivered?

Reply:
I mean is there a reason to not use map<string, VPC> vpcs (and same for interfaces and peerings? It's not that important, just checking if there was some reason